sdk/core/azure-core/src/main/java/com/azure/core/http/HttpPipeline.java

```diff
@@ -99,8 +101,23 @@ public Mono<HttpResponse> send(HttpRequest request, Context data) {
     public Mono<HttpResponse> send(HttpPipelineCallContext context) {
         // Return deferred to mono for complete lazy behaviour.
         return Mono.defer(() -> {
-            HttpPipelineNextPolicy next = new HttpPipelineNextPolicy(this, context);
+            HttpPipelineNextPolicy next =
+                new HttpPipelineNextPolicy(new HttpPipelineCallState(this, context));
             return next.process();
         });
     }
+
+    /**
+     * Wraps the request in a context with additional metadata and sends it through the pipeline.
+     *
+     * @param request THe HTTP request to send.
+     * @param data Additional metadata to pass along with the request.
+     * @return A publisher upon subscription flows the context through policies, sends the request, and emits response
+     * upon completion.
+     */
+    public HttpResponse sendSync(HttpRequest request, Context data) {
```

Comment:
Should we also have an overload that takes `HttpPipelineCallContext` like we have for async `send` method?

Reply:
for the context. 
The async callers of the `HttpPipeline` that matter call into `send(HttpRequest request, Context data)`. Some of the tests call into `send(HttpRequest request)`. Nothing but HttpPipeline itself calls into `send(HttpPipelineCallContext context)`. Last time we discussed this we were leaning towards matching just one of async overloads. We can revisit this choice if we think it would look better, but functionality wise this is all we need.